### PR TITLE
[b-10] add edit movie

### PR DESCRIPTION
**Pull request by Jokeroff**

### CODE REVIEW

File: src/main/java/com/lebediev/movieland/entity/Movie.java [hunk 1]
```diff
@@ -3,36 +3,38 @@
...
 
 public class Movie {
-    private int movieId;
-    private String movieNameRus;
-    private String movieNameNative;
-    private int date;
+    private int id;
+    private String nameRussian;
+    private String nameNative;
+    private int yearOfRelease;
```
Comment (juliamanunia) [inline src/main/java/com/lebediev/movieland/entity/Movie.java:9]: why not LocalDate?

File: src/main/java/com/lebediev/movieland/service/MovieServiceImpl.java [hunk 2]
```diff
@@ -30,9 +30,17 @@
...
 
     @Override
-    public Movie getMovieById(int movieId) {
-        return movieDao.getMovieById(movieId);
+    public Movie getMovieById(int id) {
+        return movieDao.getMovieById(id);
     }
 
+    @Override
+    public void add(Movie movie) {
```
Comment (juliamanunia) [inline src/main/java/com/lebediev/movieland/service/MovieServiceImpl.java]: how genres and countries can be added?

File: src/main/java/com/lebediev/movieland/service/MovieServiceImpl.java [hunk 3]
```diff
@@ -30,9 +30,17 @@
...
+        movieDao.add(movie);
+    }
 
+    @Override
+    public void update(Movie movie) {
+        movieDao.update(movie);
```
Comment (juliamanunia) [inline src/main/java/com/lebediev/movieland/service/MovieServiceImpl.java]: genres and countries

File: src/main/java/com/lebediev/movieland/web/controller/ReviewController.java [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+    private ReviewService reviewService;
+    @Autowired
+    private AuthService authService;
+
+    @RequestMapping(method = RequestMethod.POST)
+    @ResponseBody
+    public void add(@RequestHeader String uuid, @RequestBody String json){
+        LOG.info("Start saving review for /review for uuid: {}", uuid);
+        UserToken userToken = authService.authorize(UUID.fromString(uuid));
+        if( userToken.isUser()){
```
Comment (juliamanunia) [inline src/main/java/com/lebediev/movieland/web/controller/ReviewController.java]: admin also can add review

File: src/main/java/com/lebediev/movieland/web/controller/interceptor/Interceptor.java [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+        }
+    }
+
+    private void logRequestId(HttpServletRequest request) {
+        String uuid = request.getHeader("uuid");
+        String currentUser = "guest";
+        if (uuid != null) {
+            currentUser = authorize(uuid);
+        }
+        MDC.put("requestId", UUID.randomUUID().toString());
```
Comment (juliamanunia) [inline src/main/java/com/lebediev/movieland/web/controller/interceptor/Interceptor.java:41]: why random? User's uuid should be there

File: src/main/java/com/lebediev/movieland/web/controller/utils/JsonConverter.java [hunk 6]
```diff
@@ -2,57 +2,102 @@
...
-                return objectMapper.writerWithView(MovieViews.MovieWithReview.class).writeValueAsString(movieDto);
+                return OBJECT_MAPPER.writerWithView(MovieViews.MovieWithReview.class).writeValueAsString(movieDto);
             }
             throw new IllegalArgumentException("Wrong parameters (" + movieDto + "), jsonView = " + jsonView);
         } catch (JsonProcessingException e) {
             throw new IllegalArgumentException(e);
         }
     }
+
+    public static String toJson(TokenDto tokenDto) {
```
Comment (juliamanunia) [inline src/main/java/com/lebediev/movieland/web/controller/utils/JsonConverter.java]: to many static overloaded methods. Maybe divide a job will be better?

File: src/test/java/com/lebediev/movieland/dao/jdbc/integrationtests/JdbcMovieDaoITest.java [hunk 7]
```diff
@@ -80,10 +80,47 @@ public void testGetMoviesByGenreIdOrdered() {
...
+    }
+
+    @Test
+    public void testUpdate(){
+        Movie actual = new Movie();
+        actual.setNameNative("before update");
+        actual.setYearOfRelease(2000);
+        jdbcMovieDao.add(actual);
+        int id = jdbcTemplate.queryForObject("SELECT id FROM movie WHERE nameNative = 'before update' AND yearOfRelease = 2000", Integer.class);
+        assertNotEquals(id, null);
```
Comment (juliamanunia) [inline src/test/java/com/lebediev/movieland/dao/jdbc/integrationtests/JdbcMovieDaoITest.java]: when id will be null?

File: src/main/java/com/lebediev/movieland/web/controller/ReviewController.java (hunk 4 above)
Comment (Jokeroff) [reply src/main/java/com/lebediev/movieland/web/controller/ReviewController.java]: not requested

File: src/main/java/com/lebediev/movieland/service/MovieServiceImpl.java (hunk 2 above)
Comment (Jokeroff) [reply src/main/java/com/lebediev/movieland/service/MovieServiceImpl.java]: added in last commit

File: src/main/java/com/lebediev/movieland/web/controller/interceptor/Interceptor.java (hunk 5 above)
Comment (Jokeroff) [reply src/main/java/com/lebediev/movieland/web/controller/interceptor/Interceptor.java:41]: this is request id, not user's

File: src/main/java/com/lebediev/movieland/web/controller/utils/JsonConverter.java (hunk 6 above)
Comment (Jokeroff) [reply src/main/java/com/lebediev/movieland/web/controller/utils/JsonConverter.java]: made some simplification

File: src/test/java/com/lebediev/movieland/dao/jdbc/integrationtests/JdbcMovieDaoITest.java (hunk 7 above)
Comment (Jokeroff) [reply src/test/java/com/lebediev/movieland/dao/jdbc/integrationtests/JdbcMovieDaoITest.java]: good point - if no rows selected then exception will be thrown, fixed, thanks )

**Pull request merged**